src/cs/Ssh/IO/SshProtocol.cs

```diff
@@ -160,19 +163,25 @@ internal async Task<string> ReadProtocolVersionAsync(CancellationToken cancellat
 				if (line.StartsWith("SSH-", StringComparison.Ordinal))
 				{
 					this.metrics.AddMessageReceived(i + 1);
-					return line;
+					return (line, proxyIPAddress);
+				}
+				else if (line.StartsWith("PROXY ", StringComparison.Ordinal))
```

Comment:
Add a comment linking to the proxy protocol doc: https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt, unless you know of a more authoritative source.

Reply:
Do we need to support V2 of the PROXY protocol?